subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsTask.kt

```diff
@@ -159,7 +165,8 @@ abstract class InstrumentationTestsTask @Inject constructor(
                     registry = registry.get(),
                     reportConfig = reportConfig,
                     reportFactory = reportFactory,
-                    reportCoordinates = reportCoordinates
+                    reportCoordinates = reportCoordinates,
+                    proguardMappings = proguardMappings
```

Comment:
Can we pass Provider here?

Reply:
To do lazy calculation? Or what `Provider` do you mean?

Reply:
```kotlin
ProguardConfigurableTask {

@get:OutputFile
abstract val mappingFile: RegularFileProperty <---
```

As I see in the documentation, `WorkParameters` supports properties - https://docs.gradle.org/current/javadoc/org/gradle/workers/WorkParameters.html

`.filter { it.mappingFile.isPresent }` - this is a logic. 
I don't see reasons to spread any logic between Task and WorkAction

Reply:
Found more important reason in a discussion. 
Mappings must be input for the task.

Reply:
I've refactored this piece of code